app/scripts/services/membership/roleBindings.js

```diff
@@ -14,26 +14,12 @@ angular
 
     var roleBindingsVersion = APIService.getPreferredVersion('rolebindings');
 
-    var cache = {};
-
-    // recurse until we get a unique binding name compared to what is cached
-    var qualifyBindingName = function(name, suffix) {
-      var search = suffix ?
-                name + suffix :
-                name;
-      return _.some(cache, _.matchesProperty('metadata.name', search)) ?
-              qualifyBindingName(name, _.uniqueId()) :
-              search;
-    };
-
     var bindingTPL = function(role, projectNamespace) {
-      var roleName = _.get(role, 'metadata.name');
-      var bindingName = roleName ? qualifyBindingName(roleName) : null;
       return {
         kind: 'RoleBinding',
         apiVersion: 'v1',
         metadata: {
-          name: bindingName,
+          generateName: _.get(role, 'metadata.name'),
```

Comment:
Tested, this will ensure a unique name if the name is taken.  For example, if there is a `rolebinding` that already exists named `view` (but for some reason has bindings for `admin` or something else), this will create a name like `view12345`.

Reply:
nit: usually generateName values have a trailing `-` (for instnance, `view-` so that the names are like `view-12345`)

Reply:
Ah, funny the api doesn't handle that.  Will update.